doc/modules/prediction.rst

```diff
@@ -0,0 +1,156 @@
+.. currentmodule:: sklearn.model_selection
+
+.. _cutoffclassifier:
+
+========================================================
+Tuning cut-off decision threshold for classes prediction
+========================================================
+
+Classifiers are predictive models: they use statistical learning to predict
+outcomes. The outcomes of a classifier takes two forms: a "soft" score for each
+sample in relation to each class, and a "hard" categorical prediction (i.e.
+class label). Soft predictions are obtained using :term:`predict_proba` or
+:term:`decision_function` while hard predictions are obtained using
+:term:`predict`.
```

Comment:
Trying out some slightly different wording. Doing two comments one for this paragraph and one for the one below. I somehow don't like the "soft" and "hard". The way I think about it is that a classification model predicts a score and then we threshold that score to assign class labels.

WDYT?

```suggestion
outcomes. The output of a classifier takes two forms: a score for each
sample in relation to each class and a categorical prediction (class label).
Scores are obtained using :term:`predict_proba` or
:term:`decision_function` while class labels are obtained using
:term:`predict`.
```

Reply:
if you accept the change from "soft" and "hard" we need to fix up some later parts of the docs as well which refer to it. I can make suggestions for that if you like it, otherwise I won't spam this with useless suggestions

Reply:
I also don't like soft and hard. The statistical terms are
- `predict`: deterministic classification or decision rule
- `predict_proba`: probabilistic classification

Reply:
I see the point here. I will reformulate accordingly.

@lorentzenchr In statistical terms, how would you refer to the output of `decision_function`?

Reply:
Beware that I'm not **the** statistical reference. Here is my understanding:
`decision_function` is often nothing more than a technical detail, what matters in statistical terms is `predict_proba`. For estimators with `predict_proba`, `decision_function` is often the predictions in link space (linear predictor for linear models, raw predictions in our HGBT code). ML literatur often calls it "score" which confuses me every time.

TLDR: I don't have a good universal name for it, but it is certainly not a "decision function", our `predict` is a decision function in statistical terms.